backend/src/services/clientServerService.js

```
@@ -56,17 +62,37 @@ export async function registerClientServer(req) {
-         assigned_schema_name,
-         allowed_return_urls,
-         user_id: null, // Public registration has no user
-         client_mode: "api-auth-server", // Default for public API
-      });
+      if (user_email_for_id) {
+         const user = await userRepo.getUserByEmail(
+            pool,
+            "auth_internal",
+            user_email_for_id
+         );
+         if (user && user.id) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)